src/sentry/api/serializers/rest_framework/mentions.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import absolute_import
+
+from rest_framework import serializers
+
+from sentry.api.fields.actor import Actor
+from sentry.models import (
+    Team,
+    User,
+)
+
+
+def extract_user_ids_from_mentions(organization_id, mentions):
+    """
+    Extracts user ids from a set of mentions. Mentions should be a list of
+    `Actor` instances. Returns a dictionary with 'users' and 'team_users' keys.
+    'users' is the user ids for all explicitly mentioned users, and 'team_users'
+    is all user ids from explicitly mentioned teams, excluding any already
+    mentioned users.
+    """
+    actors = Actor.resolve_many(mentions)
+    actor_mentions = seperate_resolved_actors(actors)
+
+    mentioned_team_users = list(
+        Team.objects.get_users_from_teams(
+            organization_id,
+            actor_mentions['teams'],
+        ).exclude(id__in={u.id for u in actor_mentions['users']}).values_list(
+            'id',
+            flat=True,
+        )
+    )
+
+    return {
+        'users': set([user.id for user in actor_mentions['users']]),
+        'team_users': set(mentioned_team_users),
+    }
+
+
+def seperate_actors(actors):
+    users = [actor for actor in actors if actor.type is User]
+    teams = [actor for actor in actors if actor.type is Team]
+
+    return {'users': users, 'teams': teams}
+
+
+def seperate_resolved_actors(actors):
+    users = [actor for actor in actors if isinstance(actor, User)]
+    teams = [actor for actor in actors if isinstance(actor, Team)]
+
+    return {'users': users, 'teams': teams}
+
+
+class MentionsMixin(object):
+    def validate_mentions(self, attrs, source):
+        if source in attrs and 'group' in self.context:
+
+            mentions = attrs[source]
+            seperated_actors = seperate_actors(mentions)
+            # Validate that all mentioned users exist and are on the project.
```

Comment:
Does this still work if there are multiple projects like in incidents?

Reply:
Damn, completely missed this...